### DISCUSSION TITLE
Issue running upgrade.sh

### DISCUSSION THREAD

**Discussion by billraff** [jasonacox/Powerwall-Dashboard#398] [Q&A]
I run Powerwall Dashboard on my Synology. I wanted to check for updates so tried to run the upgrade.sh script. It gave some errors so based on another discussion thread in this forum I installed from the Synocommunity the Git package. 
Now when I run upgrade.sh  this is what I see -

billr@DS220:/volume1/powerwall$ ./upgrade.sh
Upgrade Powerwall-Dashboard from 2.9.10 to 3.0.2
---------------------------------------------------------------------
This script will attempt to upgrade you to the latest version without
removing existing data. A backup is still recommended.

Upgrade - Proceed? [y/N] y

Resetting Timezone to Default...

Pull influxdb.sql, dashboard.json, telegraf.conf, and other changes...

You do not have the initial commit yet

So I don't know anything about Git so not sure how I get the initial commit to allow this script to continue on its way.
Thanks in advance for any advice.

**Comment by jeffkruczynski** [comment]
FWIW, since you were having the same Git-related issue I was, I just ran the upgrade script as well, from 2.10.0 to 3.0.2. It seems to have upgraded fine (no issues during the upgrade) although the Power Flow pane does not seem to want to update. Sometimes that happens, so I'll give it a little time before I decide it's a problem with the upgrade.
But in any event: my point is that the error you're running into doesn't necessarily seem to be caused by the upgrade itself.

How did you do the initial Powerwall-Dashboard install?

**Reply by jeffkruczynski** [reply]
I spoke too soon...the pypowerwall container seems to be stopping unexpectedly now and not collecting data. Working on it.

**Reply by jeffkruczynski** [reply]
This appears resolved & was related to the [firmware upgrade discussion](https://github.com/jasonacox/Powerwall-Dashboard/discussions/367).

**Reply by billraff** [reply]
I initially did the install using the instructions on this repository. Cloned it and ran the setup script. After that I actually uninstalled the Git package since I thought all future updates would be done via Synology's Container Manager. I reinstalled the Git package from the Synocommunity. I then, just to try and get things current, ran -

    git clone https://github.com/jasonacox/Powerwall-Dashboard.git

thinking it would pull the latest code. But when running the ./upgrade.sh I'm left with the error -

Pull influxdb.sql, dashboard.json, telegraf.conf, and other changes...

You do not have the initial commit yet

So stuck there. I'm guessing it's git that needs something issued in order to do this 'initial commit' but darned if I know what.

**Reply by jasonacox** [reply]
Ah, that explains it. You never ran `setup.sh` so you won't be able to run `upgrade.sh`. Are you able to run `setup.sh`?

**Comment by billraff** [comment]
Yes, I can run ./setup.sh successfully. Just did it and it recognized my previous settings and went through without error. After it finished I ran the upgrade script and received the same error:

Upgrade Powerwall-Dashboard from 2.9.10 to 3.0.3
---------------------------------------------------------------------
This script will attempt to upgrade you to the latest version without
removing existing data. A backup is still recommended.

Upgrade - Proceed? [y/N] y

Resetting Timezone to Default...

Pull influxdb.sql, dashboard.json, telegraf.conf, and other changes...

You do not have the initial commit yet


Frustrating but I'm stubborn. Is there a proper uninstall process where I can just start from scratch? Would it be as simple as stopping the containers and deleting then deleting the contents of the Powerwall folder on my NAS?

**Reply by jasonacox** [reply]
> I'm guessing it's git that needs something issued in order to do this 'initial commit' but darned if I know what.

I think you are right. I suspect the error is coming from `git stash` in the upgrade script.  Some things to try:

```bash
# First, what how does git report on status?
git status

# Attempt to create an initial commit we can stash
echo "Temp" > temp
git add temp
git commit -m "stash first commit"
git stash

# Try again
git status
./upgrade.sh
```

> just start from scratch? 

Yes, you could. I actually have a script that does this as part of my test before a release:

1. Archive the database (so you preserve the database)
2. Delete the Powerwall-Dashboard directory
3. Run `git clone` and run `setup.sh` from scratch
5. Restore archive of database to bring back the history

```bash
#!/bin/bash

echo "Reinstall Powerwall-Dashboard"

read -r -p "Are you sure? [y/N] " response
case "$response" in
    [yY][eE][sS]|[yY]) 
        echo "Here we go..."
        ;;
    *)
        echo "Exit"	
        exit 0
        ;;
esac

# Archive
BACKUP=`date +"Powerwall-Dashboard.%Y-%m-%d.tgz"`
echo "Archiving to $BACKUP..."
cd Powerwall-Dashboard
./compose-dash.sh down
cd ..
sudo tar -zcf $BACKUP Powerwall-Dashboard/
echo "Taking database snapshot..."
sudo tar -zcf snapshot.tgz Powerwall-Dashboard/influxdb/

# Optional - Clean up old containers and file system
echo "Stopping and Cleaning up old containers..."
docker-compose -f Powerwall-Dashboard/powerwall.yml rm
docker rm pypowerwall
docker images | grep pypowerwall | awk '{print $3}' | xargs docker rmi -f
docker rm telegraf
docker images | grep telegraf | awk '{print $3}' | xargs docker rmi -f
docker rm grafana
docker images | grep grafana | awk '{print $3}' | xargs docker rmi -f
docker rm influxdb
docker images | grep influxdb | awk '{print $3}' | xargs docker rmi -f
docker rm weather411
docker images | grep weather411 | awk '{print $3}' | xargs docker rmi -f

# Delete old install
echo "Deleting old install..."
sudo rm -fr Powerwall-Dashboard

# Clone new
echo "Cloning from github..."
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
cd Powerwall-Dashboard
pwd

# Run setup
./setup.sh

# Restore database
echo "Restoring database snapshot..."
docker stop influxdb
cd ..
sudo rm -fr Powerwall-Dashboard/influxdb/
sudo tar -zxf snapshot.tgz
docker start influxdb
```

**Comment by billraff** [comment]
Well all is finally well. When I originally set up this application 3 or 4 months ago I set up on a Raspberry pi 4 I had that was gathering dust. Plugged it in to see if it would upgrade successfully. Rpi would not boot up which proves the old adage about raining and then pouring. So took a hammer to the Synology build and got it back up and running there. Tested the upgrade script and it now runs successfully. Thanks all for your help.

**Reply by jasonacox** [reply]
That's great news @billraff !  In case someone else has the same problem, what fixed your issue?  Did you do a full rebuild or did the first "commit" test work?

**Reply by billraff** [reply]
The commit test failed. It said something about trying to use the 'add' command first. At that point I decided to remove and reinstall into a new directory on the NAS.
